hub-detect/src/main/resources/hub-detect-sh

```diff
@@ -57,63 +66,53 @@ get_detect() {
+      DETECT_RELEASE_VERSION=$(curl -sSL $DETECT_CURL_OPTS 'https://test-repo.blackducksoftware.com/artifactory/api/search/latestVersion?g=com.blackducksoftware.integration&a=hub-detect&repos=bds-integrations-release')
       DETECT_SOURCE="https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/${DETECT_RELEASE_VERSION}/hub-detect-${DETECT_RELEASE_VERSION}.jar"
       DETECT_DESTINATION="${DETECT_JAR_PATH}/hub-detect-${DETECT_RELEASE_VERSION}.jar"
     else
       DETECT_SOURCE="https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/${DETECT_RELEASE_VERSION}/hub-detect-${DETECT_RELEASE_VERSION}.jar"
       DETECT_DESTINATION="${DETECT_JAR_PATH}/hub-detect-${DETECT_RELEASE_VERSION}.jar"
     fi
-    echo "will look for : ${DETECT_SOURCE}"
+    echo "will look for release: ${DETECT_SOURCE}" >&2
   fi
 
   USE_REMOTE=1
   if [ "$CURRENT_VERSION" != "$LATEST_VERSION" ] && [ $DETECT_USE_SNAPSHOT -eq 1 ]; then
-    echo "You don't have the latest snapshot, so the new snapshot will be downloaded."
+    echo "You don't have the latest snapshot, so the new snapshot will be downloaded." >&2
   elif [ ! -f $DETECT_DESTINATION ]; then
-    echo "You don't have the current file, so it will be downloaded."
+    echo "You don't have the current file, so it will be downloaded." >&2
   else
-    echo "You have already downloaded the latest file, so the local file will be used."
+    echo "You have already downloaded the latest file, so the local file will be used." >&2
     USE_REMOTE=0
   fi
 
   if [ $USE_REMOTE -eq 1 ]; then
-    echo "getting ${DETECT_SOURCE} from remote"
-    curl $DETECT_CURL_OPTS -L -o $DETECT_DESTINATION "${DETECT_SOURCE}"
-    echo "saved ${DETECT_SOURCE} to ${DETECT_DESTINATION}"
+    echo "getting ${DETECT_SOURCE} from remote" >&2
+    curl -sSL $DETECT_CURL_OPTS -o "${DETECT_DESTINATION}" "${DETECT_SOURCE}"
+    echo "saved ${DETECT_SOURCE} to ${DETECT_DESTINATION}" >&2
   fi
 }
 
 run_detect() {
-  JAVACMD="java ${DETECT_JAVA_OPTS} -jar ${DETECT_DESTINATION}"
-  echo "running detect: ${JAVACMD} ${LOGGABLE_SCRIPT_ARGS}"
-
-  # first, silently delete (-f ignores missing
-  # files) any existing shell script, then create
-  # the one we will run
-  rm -f $DETECT_JAR_PATH/hub-detect-java.sh
-  echo "#!/bin/sh" >> $DETECT_JAR_PATH/hub-detect-java.sh
-  echo "" >> $DETECT_JAR_PATH/hub-detect-java.sh
-  echo $JAVACMD $SCRIPT_ARGS >> $DETECT_JAR_PATH/hub-detect-java.sh
-  source $DETECT_JAR_PATH/hub-detect-java.sh
-  RESULT=$?
-  echo "Result code of ${RESULT}, exiting"
-  rm -f $DETECT_JAR_PATH/hub-detect-java.sh
-  exit $RESULT
+  javacmd=(java ${DETECT_JAVA_OPTS} -jar "${DETECT_DESTINATION}")
+  type -a java || return -1
+  echo "running detect: ${javacmd[*]@Q} ${loggable_script_args[*]@Q}" >&2
+  "${javacmd[@]}" "${script_args[@]}" || return $?
```

Comment:
Is it a valid condition to have special characters within these 2 variables which require the need to print them with '@Q'?

Reply:
It quotes array elements, according to the new "parameter transformation" man paragraph,

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html